crates/ruma-events/src/direct.rs

```diff
@@ -80,20 +139,43 @@ mod tests {
 
     #[test]
     fn deserialization() {
-        let alice = owned_user_id!("@alice:ruma.io");
+        let alice = DirectUserIdentifier::from_borrowed("@alice:ruma.io");
         let rooms = vec![owned_room_id!("!1:ruma.io"), owned_room_id!("!2:ruma.io")];
 
         let json_data = json!({
             "content": {
-                alice.to_string(): rooms,
+                alice: rooms,
+                "alice@ruma.io": vec![owned_room_id!("!3:ruma.io")],
             },
             "type": "m.direct"
         });
 
         let event: DirectEvent = from_json_value(json_data).unwrap();
-        let direct_rooms = event.content.get(&alice).unwrap();
+        let direct_rooms = event.content.get(alice).unwrap();
```

Comment:
For ease of use with `(Owned)UserId`, I am wondering if we should implement `Borrow<UserId> for OwnedDirectUserIdentifier`, or maybe that is too far-fetched.

Reply:
Ah but we can't, so nevermind.